### PR TITLE
kubeadm: add a kind based job under sig-cluster-lifecycle

### PR DESCRIPTION
**Pull request by neolit123**
/priority important-soon
/kind feature
/assign @BenTheElder @krzyzacy 
cc @timothysc @fabriziopandini

### CODE REVIEW

**Comment by neolit123** [comment]
need to update my bazel locally.

**Comment by krzyzacy** [comment]
/approve
@BenTheElder to double check the flags

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      # skip serial tests and run with --ginkgo-parallel
+      - name: "PARALLEL"
+        value: "true"
+      args:
+      - "--job=$(JOB_NAME)"
+      - "--root=/go/src"
+      - "--repo=k8s.io/kubernetes=master"
+      - "--repo=k8s.io/kubeadm=master"
+      - "--repo=sigs.k8s.io/kind=master"
+      - "--service-account=/etc/service-account/service-account.json"
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: do we need the service account?

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+periodics:
+- name: ci-kubernetes-e2e-kubeadm-kind-master
+  interval: 1h
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: is this rate OK?

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      - "--service-account=/etc/service-account/service-account.json"
+      - "--upload=gs://kubernetes-jenkins/logs"
+      - "--scenario=kubernetes_e2e"
+      - "--timeout=30m"
+      - "--"
+      - "--cluster="
+      - "--build=bazel"
+      - "--provider=skeleton"
+      - "--deployment=kind"
+      - "--kind-binary-version=latest"
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: using this until we make a new release that includes some of the recent CLI features we've added e.g. `kind get nodes`

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      - "--kind-config./../../k8s.io/kubeadm/tests/e2e/kind/single-cp/single-cp.yaml"
+      - "--test-cmd=./../../k8s.io/kubeadm/tests/e2e/kind/test-cmd-basic.sh"
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: these are here: https://github.com/kubernetes/kubeadm/tree/master/tests/e2e/kind

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      hostPath:
+        path: /sys/fs/cgroup
+        type: Directory
+    # trialing this on kind jobs, we are using FQDN for in-cluster services, now
+    # so use ndots 1 to improve dns performance
+    # TODO(bentheelder): consider setting this at the cluster level instead
+    dnsConfig:
+      options:
+        - name: ndots
+          value: "1"
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: ^ the rest is just a copy-paste. my assumption is that we are going to need it for all jobs?

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml (hunk 4 above)
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: so --test-cmd runs kubetest itself recursively in this case, @krzyzacy thoughts?

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml (hunk 5 above)
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: we can ditch the ndots, it makes the config more verbose and results have been inconclusive imho. it's difficult to measure

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml (hunk 4 above)
Comment (krzyzacy) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: which should be... fine?

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml (hunk 1 above)
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: bootstrap.py needs this to do log uploads iirc. it's hidden away when doing a pod-utils job

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      - "--provider=skeleton"
+      - "--deployment=kind"
+      - "--kind-binary-version=latest"
+      - "--kind-config./../../k8s.io/kubeadm/tests/e2e/kind/single-cp/single-cp.yaml"
+      - "--test-cmd=./../../k8s.io/kubeadm/tests/e2e/kind/test-cmd-basic.sh"
+      - "--timeout=30m"
+      # we need privileged mode in order to do docker in docker
+      securityContext:
+        privileged: true
+      # kind needs /lib/modules and cgroups from the host
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: note to self: create a preset for this. unless of course, we're going to delete those...

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+      volumeMounts:
+      - mountPath: /lib/modules
+        name: modules
+        readOnly: true
+      - mountPath: /sys/fs/cgroup
+        name: cgroup
+      resources:
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: eventually what we should probably do is build & push a CI image periodically and then have N jobs consume the latest image, but that will come later.

**Comment by neolit123** [comment]
thanks a lot for the comments. will address later today!

**Comment by timothysc** [comment]
@neolit123 - so the only thing that the other -master job does now is to verify that we are building packages?

**Comment by neolit123** [comment]
we need a separate job for package "install testing" for debs and rpms.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml (hunk 4 above)
Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: :crossed_fingers:

**Comment by neolit123** [comment]
PR updated

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml (hunk 3 above)
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: soon! 😅 
maybe add a TODO?

Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: i will not forget to update this for all the jobs that will be added before that,.

**Comment by neolit123** [comment]
/hold cancel
i will cancel the hold so that i can observe some runs today. :roll_eyes:

**Comment by neolit123** [comment]
it seems like this is failing with
```
W0207 15:30:21.354] IOError: [Errno 2] No such file or directory: '/root/.ssh/google_compute_engine'
```
trying to figure out why this ends up as `/root/.ssh`
https://github.com/kubernetes/test-infra/blob/master/scenarios/kubernetes_e2e.py#L228

my understanding is that this should be the workspace path instead of `/root`.
so i wonder what is missing.

**Comment by BenTheElder** [comment]
@neolit123 imo we should write a pod-utils job with a bare kubetest call instead and ditch the scenario adding nonsense on top. We don't need things like the SSH key at all.

**Pull request merged**